src/main/java/appeng/me/storage/MEMonitorIFluidHandler.java

```diff
@@ -0,0 +1,327 @@
+
+package appeng.me.storage;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.NavigableMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+
+import appeng.api.AEApi;
+import appeng.api.config.AccessRestriction;
+import appeng.api.config.Actionable;
+import appeng.api.config.StorageFilter;
+import appeng.api.networking.security.IActionSource;
+import appeng.api.networking.ticking.TickRateModulation;
+import appeng.api.storage.IMEMonitor;
+import appeng.api.storage.IMEMonitorHandlerReceiver;
+import appeng.api.storage.IStorageChannel;
+import appeng.api.storage.channels.IFluidStorageChannel;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.api.storage.data.IItemList;
+
+
+public class MEMonitorIFluidHandler implements IMEMonitor<IAEFluidStack>, ITickingMonitor
+{
+	private final IFluidHandler handler;
+	private final IItemList<IAEFluidStack> list = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+	private final HashMap<IMEMonitorHandlerReceiver<IAEFluidStack>, Object> listeners = new HashMap<>();
+	private final NavigableMap<Integer, CachedFluidStack> memory;
+	private IActionSource mySource;
+	private StorageFilter mode = StorageFilter.EXTRACTABLE_ONLY;
+
+	public MEMonitorIFluidHandler( final IFluidHandler handler )
+	{
+		this.handler = handler;
+		this.memory = new ConcurrentSkipListMap<>();
+	}
+
+	@Override
+	public void addListener( final IMEMonitorHandlerReceiver<IAEFluidStack> l, final Object verificationToken )
+	{
+		this.listeners.put( l, verificationToken );
+	}
+
+	@Override
+	public void removeListener( final IMEMonitorHandlerReceiver<IAEFluidStack> l )
+	{
+		this.listeners.remove( l );
+	}
+
+	@Override
+	public IAEFluidStack injectItems( final IAEFluidStack input, final Actionable type, final IActionSource src )
+	{
+		final int filled = this.handler.fill( input.getFluidStack(), type == type.MODULATE );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2209)